src/easier68k/core/models/assembly_parameter.py

```diff
@@ -0,0 +1,219 @@
+from ..util.conversions import to_word
+
+# should try to make this a constant only defined once
+MAX_MEMORY_LOCATION = 16777216  # 2^24
+
+
+class AssemblyParameter:
+
+    def __init__(self, mode: EAMode = EAMode.ERR, data: int = 0, length: int = 4):
+        """
+        Constructor
+
+        Checks that the mode and the data are considered to be valid, like
+        registers are within the correct bounds and addresses are not out of bounds.
+        """
+
+
+        # ensure that the values are valid
+
+        # when referencing a register, ensure that the data is within [0, 7]
+        if mode in [EAMode.DataRegisterDirect, EAMode.AddressRegisterIndirectPreDecrement,
+                    EAMode.AddressRegisterIndirect, EAMode.AddressRegisterIndirectPostIncrement,
+                    EAMode.AddressRegisterDirect]:
+            assert 0 <= data <= 7, 'The register number for this mode must be in the range [0, 7]!'
+
+        if mode in [EAMode.AbsoluteWordAddress, EAMode.AbsoluteLongAddress]:
+            # ensure that the address isn't outside the bounds of max memory location
+            # and is greater than or eq to 0
+            assert 0 <= data <= MAX_MEMORY_LOCATION, 'An absolute address must be in the bounds [0, 2^24]!'
+
+        if mode is EAMode.Immediate:
+            # ensure that the value isn't too large, cannot be larger than a long word
+            # negative values need to be converted into unsigned integers
+            assert 0 <= data <= 0xFFFFFFFF, 'Value must fit inside a long word!'
+
+        assert mode is not EAMode.ERR, 'EAMode cannot be ERROR.'
+
+        assert length == 1 or length == 2 or length == 4, 'Length must 1 (byte) 2 (word) 4 (long)'
+
+        # set values
+        self.mode = mode
+        self.data = data
+        self.length = length
+
+    def __str__(self):
+        """
+        str util method
+        :return:
+        """
+        return "EA Mode: {}, Data: {}".format(self.mode, self.data)
+
+    def get_value(self, simulator: M68K) -> int:
+        """
+        Gets the value for this EAMode
+        :param simulator:
+        :return:
+        """
+        if self.mode is EAMode.IMM:
+            return self.data
+
+        if self.mode is EAMode.DRD:
+            # convert the data into the register value
+            assert 0 <= self.data <= 7
+            data_register = Register(self.data)
+            return simulator.get_register_value(data_register)
+
+        if self.mode is EAMode.AddressRegisterDirect:
+            # address register direct gets the value of the register
+            assert Register.A0 <= self.data + Register.A0 <= Register.A7
+            # offset the value to compensate for the enum offset
+            addr_register = Register(self.data + Register.A0)
+            # get the value of the register, that's it
+            return simulator.get_register_value(addr_register)
+
+        if self.mode is EAMode.AddressRegisterIndirect:
```

Comment:
To remove some redundant code (personal preference), you could merge these two if statements and then have a nested one inside (saves a repetitive if statement)